draft-ietf-quic-qlog-main-schema.md

```diff
@@ -1137,18 +1143,72 @@ SimulationMarker = {
 ~~~
 {: #simulation-marker-def title="SimulationMarker definition"}
 
-# Event definition guidelines
+# Event extensibility {#event-extensibility}
+
+The main qlog schema defined by this document describes log and trace formats
+that have related events. A minimal set of common events are presented in
+{{events}}. It is expected that logging is extended with specific, per-protocol
+event definitions that specify the name (category + type) and data needed for
+each individual event. Examples include the QUIC event definitions {{QLOG-QUIC}}
+and HTTP/3 event definitions {{QLOG-H3}}.
+
+New event definitions SHOULD follow the guidance in this section, which provides
+consistency that makes it easier for qlog implementers to support a wide variety
+of protocols.
+
+## Extended Event Schema
+
+New event definitions SHOULD be part of an extension schema, using the
+annotations and concepts presented in this section.
+
+A schema defines one or more qlog event categories that can contain multiple
+event definitions. Each event MUST only belong to a single category.
 
-This document defines the main schema for the qlog format together with some
-common events, which on their own do not provide much logging utility. It is
-expected that logging is extended with specific, per-protocol event definitions
-that specify the name (category + type) and data needed for each individual
-event. Examples include the QUIC event definitions {{QLOG-QUIC}} and HTTP/3
-event definitions {{QLOG-H3}}.
+It is not possible to extend a category with new events. Instead, extension
+schema MUST define new categories with globally unique identifiers. Per
+{{{#name-field}}}, event names are the concatenation of category and type.
+Therefore, unique category identifiers ensure events in qlog files are
+disambuiguated.
+
+Each extension schema is named by a URI. That URI MUST be absolute; it precisely
+identifies the format and meaning of the extension. URIs that contain a domain
+name SHOULD also contain a month-date in the form mmyyyy. The definition of the
+schema and assignment of the URI MUST have been authorized by the owner of the
+domain name on or very close to that date. (This avoids problems when domain
+names change ownership.) If the resource or document defines several event
+categories, then the URI MUST identify the actual category in use, e.g., using
+a fragment or query identifier (characters after a "#" or "?" in the URI).
+
+For extensions defined in RFCs, the URI used SHOULD be a URN starting with
+`urn:ietf:params:qlog:` followed by a registered, descriptive name.
+
+A log file that uses events from extension schema SHOULD list all schema
+identifiers in the `additional_event_schemas` field; see {{qlog-file-schema}}
```

Comment:
```suggestion
identifiers (potentially multiple per extension schema in the case multiple categories are used) in the `additional_event_schemas` field; see {{qlog-file-schema}}
```

I feel that at this point, there's lots of ambiguity between a "schema identifier" and a specification of it for a specific category and what should actually be listed in the `additional_event_schemas`. The example below makes it a bit clearer, but I think we need more in this prose as well. I think the above is the minimum we could get away with, but potentially more text is needed (and maybe a new term like "category identifier"?)

Reply:
See new text